fault-remediation/pkg/reconciler/reconciler.go

```diff
@@ -68,41 +74,52 @@ func NewReconciler(cfg ReconcilerConfig, dryRunEnabled bool) *Reconciler {
 }
 
 func (r *Reconciler) Start(ctx context.Context) error {
-	watcher, err := storewatcher.NewChangeStreamWatcher(ctx, r.Config.MongoConfig, r.Config.TokenConfig,
-		r.Config.MongoPipeline)
+	// Create datastore instance
+	ds, err := datastore.NewDataStore(ctx, r.Config.DataStoreConfig)
 	if err != nil {
-		return fmt.Errorf("error initializing change stream watcher: %w", err)
+		return fmt.Errorf("error initializing datastore: %w", err)
 	}
 
 	defer func() {
-		if err := watcher.Close(ctx); err != nil {
-			slog.Error("failed to close watcher", "error", err)
+		if err := ds.Close(ctx); err != nil {
+			slog.Error("failed to close datastore", "error", err)
 		}
 	}()
 
-	collection, err := storewatcher.GetCollectionClient(ctx, r.Config.MongoConfig)
-	if err != nil {
-		slog.Error("error initializing collection client for mongodb",
-			"config", r.Config.MongoConfig,
-			"error", err)
+	// Create watcher using the factory pattern
+	watcherConfig := watcher.WatcherConfig{
+		Pipeline:       r.Config.Pipeline,
+		CollectionName: "HealthEvents",
+	}
 
-		return fmt.Errorf("error initializing collection client for mongodb: %w", err)
+	watcherInstance, err := watcher.CreateChangeStreamWatcher(ctx, ds, watcherConfig)
```

Comment:
ideally we could have the same `CreateChangeStreamWatcher` function used across `fault-quarantine` and `fault-remediation`. Using the same function would force more consistent in the params as well across the sub projects and might help later if we want to standardized to a common reconciler/watcher thing.